mne/tests/test_epochs.py

```diff
@@ -1511,50 +1511,102 @@ def test_split_saving(tmp_path, split_size, n_epochs, n_files, size, metadata, c
             }
         )
         epochs.metadata = metadata
```

Comment:
I don't see, how the metadata are used in the tests.  Do metadata affect `epochs.drop_bad()`? I don't think they do. Same thing with `concat`. Maybe they should be removed.

Reply:
They were probably added to make sure that we actually split the metadata properly (or copy it? not sure which we do currently) when splitting the epoch data across multiple files

Reply:
I've looked into commit history and it seems they are added to fix #7897. My current understanding is that there was a bug with saving splits when the size just marginally exceeded the splitting threshold, so metadata and concat manipulations help catch this corner case. Does this make sense?

Also there's #5102 related to problem with events when loading back the splits.

Reply:
thx:)